src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -645,6 +645,22 @@ private static Expression CreateArgument(ParameterInfo parameter, RequestDelegat
             throw new InvalidOperationException($"Encountered a parameter of type '{parameter.ParameterType}' without a name. Parameters must have a name.");
         }
 
+        if (parameter.ParameterType.IsByRef)
+        {
+            var attribute = string.Empty;
+
+            if (parameter.Attributes.HasFlag(ParameterAttributes.In))
+            {
+                attribute = "in ";
+            }
+            else if (parameter.Attributes.HasFlag(ParameterAttributes.Out))
+            {
+                attribute = "out ";
+            }
```

Comment:
Should there be a "ref" alternative here?

Reply:
Probably. Should that just be the default instead of ""?

Reply:
I added "by reference" in the error message but why not.  Let me update it.